python/teams_ai/planner/planner.py

```diff
@@ -0,0 +1,39 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from abc import ABC, abstractclassmethod
+from typing import Any, Callable, Union
+from teams_ai.turn_state import TurnState
+
+class Planner(ABC):
+    """
+    interface implemented by all planners
+    """
+
+    @abstractclassmethod
+    async def generate_plan(self, state: TurnState, prompt_name_or_template: Union[str, PromptTemplate], *, history_options: AIHistoryOptions) -> Plan:
```

Comment:
How to get the prompt given the prompt name?

Reply:
The planner implementation will have its own prompt manager, it will take care of loading the prompt.